src/main/java/io/vertx/junit5/VertxExtension.java

```diff
@@ -46,12 +53,48 @@ public boolean supportsParameter(ParameterContext parameterContext, ExtensionCon
     return type == VertxTestContext.class || type == Vertx.class;
   }
 
+  private void awaitCheckpoints(ExtensionContext extensionContext) throws Exception {
+    if (extensionContext.getExecutionException().isPresent()) {
+      return;
+    }
+    VertxTestContext context = store(extensionContext).remove("VertxTestContext", VertxTestContext.class);
+    if (context != null) {
+      int timeoutDuration = DEFAULT_TIMEOUT_DURATION;
+      TimeUnit timeoutUnit = DEFAULT_TIMEOUT_UNIT;
+      Optional<Method> testMethod = extensionContext.getTestMethod();
+      if (testMethod.isPresent() && testMethod.get().isAnnotationPresent(Timeout.class)) {
+        Timeout annotation = extensionContext.getRequiredTestMethod().getAnnotation(Timeout.class);
+        timeoutDuration = annotation.value();
+        timeoutUnit = annotation.timeUnit();
+      } else if (extensionContext.getRequiredTestClass().isAnnotationPresent(Timeout.class)) {
+        Timeout annotation = extensionContext.getRequiredTestClass().getAnnotation(Timeout.class);
+        timeoutDuration = annotation.value();
+        timeoutUnit = annotation.timeUnit();
+      }
+      if (context.awaitCompletion(timeoutDuration, timeoutUnit)) {
+        if (context.failed()) {
+          Throwable throwable = context.causeOfFailure();
+          if (throwable instanceof Exception) {
```

Comment:
I wonder if this should be done the other way like:

```java
if (throwable instanceof AssertionError) {
   throw (AssertionError) throwable;
} else {
   throw new AssertionError(throwable);
}
```

And it looks strange to me but I might be reading it wrong... :) 

Just wondering if someone throws a Throwable or any Throwable subclass (not an Exception or any Exception subclass).

Reply:
Well the thing is that the JUnit5 extension API forces the method to have a `throws Exception` declaration, so we can't _simply_ throw the exception as-it, hence we have to wrap unchecked exceptions into checked exceptions.

Makes sense? (or am I wrong?)

Reply:
why unchecked does not work ? I'm affraid it translates a JUnit failure into an assertion error and lose contextual information